File: src/client/Phone.java
package client;

import org.json.JSONObject;

public final class Phone extends Device {

    private int x, y;

    public Phone() {
        super("Jamie");
        this.x = 0;
        this.y = 0;
    }

    @Override
    protected void performTask() {
        System.out.println("x + y = " + (this.x + this.y));
    }

    @Override
    protected boolean update(JSONObject json) {
        int newX, newY;
        try {
            newX = (Integer) json.get("x");
            newY = (Integer) json.get("y");
            this.x = newX;
            this.y = newY;
            return true;
        } catch (Exception e) {
            return false;
        }
    }

    public static void main(String [] args) {
        Device phone = new Phone("69 LOL", "0.1.5");
        // Connect to the server
        phone.connectDevice();
        // Apply for a ticket
        phone.applyForTicket();
        if(phone.isUpdateAvailable() && phone.getUpdateFile()) {
            // implement bspatch so we can install the update patch file
            phone.applyPatch();
        }
    }
}

File: src/client/Phone.py
from Device import Device
import time
import sys

class Phone(Device):
    def __init__(self, id, version):
        super().__init__(id, version)

    def main(self):
        

        print("Gonna connect this phone!")
        self.connectDevice()
        
        while True:
            print("Connected! Lets check for an update")
            while(not self.isUpdateAvailable()):
                print("doing my own things. no update available. id "+ self.id)
                time.sleep(3)
            
            print("I need one! Lets get a ticket!"+ self.id)
            while(not self.applyForTicket()):
                time.sleep(2)
            
            print("Got a ticket! Gonna apply it"+ self.id)
            if self.getUpdateFile():
                print("got the update. applying it"+ self.id)
                self.applyPatch()
                print("update done"+ self.id)
                self.confirmUpdate()
                print("Everything worked!"+ self.id)
                



if __name__ == "__main__":

    deviceId = sys.argv[1]

    phone = Phone(deviceId, 1)


    phone.main()
